File: java-advent-of-code/src/test/java/advent/twenty_fifteen/Day24ItHangsInTheBalanceTest.java
package advent.twenty_fifteen;

import advent.common.DailyProblem;
import org.junit.Test;

import static org.junit.Assert.assertEquals;

public class Day24ItHangsInTheBalanceTest {
    @Test
    public void checkReal() {
        DailyProblem<Long, Long> dp = new Day24ItHangsInTheBalance("/twenty_fifteen/Day24-ItHangsInTheBalance-input.txt");

        assertEquals(10439961859l, dp.getPart1Answer().longValue());
        assertEquals(72050269l, dp.getPart2Answer().longValue());
    }

    @Test
    public void checkExample1() {
        DailyProblem<Long, Long> dp = new Day24ItHangsInTheBalance("/twenty_fifteen/Day24-ItHangsInTheBalance-example#1.txt");

        assertEquals(88, dp.getPart1Answer().longValue());
        assertEquals(33, dp.getPart2Answer().longValue());
    }
}

File: scala-advent-of-code/src/test/scala/advent/twenty_fifteen/Day24ItHangsInTheBalanceTest.scala
package advent.twenty_fifteen

import org.scalatest.{FlatSpec, Matchers}

class Day24ItHangsInTheBalanceTest  extends FlatSpec with Matchers {
  "2015 Day 24 - Input" should "calculate answers" in {
    val dp = new Day24ItHangsInTheBalance("/twenty_fifteen/Day24-ItHangsInTheBalance-input.txt")
    dp.part1Answer should be(10439961859L)
    dp.part2Answer should be(72050269L)
  }

  "2015 Day 24 - Example #1" should "calculate answers" in {
    val dp = new Day24ItHangsInTheBalance("/twenty_fifteen/Day24-ItHangsInTheBalance-example#1.txt")
    dp.part1Answer should be(88)
    dp.part2Answer should be(33)
  }
}
